src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -425,15 +425,12 @@ protected ODataJsonLightReaderNestedResourceInfo InnerReadUndeclaredProperty(IOD
                 ValidateExpandedNestedResourceInfoPropertyValue(this.JsonReader, isCollection, propertyName, payloadTypeReference);
                 if (isCollection)
                 {
-                    readerNestedResourceInfo = this.ReadingResponse
-                        ? ReadExpandedResourceSetNestedResourceInfo(resourceState, null, payloadTypeReference.ToStructuredType(), propertyName, /*isDeltaResourceSet*/ false)
-                        : ReadEntityReferenceLinksForCollectionNavigationLinkInRequest(resourceState, null, propertyName, /*isExpanded*/ true);
+                    readerNestedResourceInfo =
+                        ReadExpandedResourceSetNestedResourceInfo(resourceState, null, payloadTypeReference.ToStructuredType(), propertyName, /*isDeltaResourceSet*/ false);
                 }
                 else
                 {
-                    readerNestedResourceInfo = this.ReadingResponse
-                        ? ReadExpandedResourceNestedResourceInfo(resourceState, null, propertyName, payloadTypeReference.ToStructuredType(), this.MessageReaderSettings)

```

Comment:
Are "ReadExpandedResourceNestedResourceInfo"
 and "ReadEntityReferenceLinksForCollectionNavigationLinkInRequest" still used, or can those methods be deleted?

Reply:
@mikepizzo I think `ReadExpandedResourceNestedResourceInfo` is still being used

Reply:
The change is only in 'InnerReadUndeclaredProperty', We are in the method because It's "Undeclared" property.
For normal property, 'ReadEntityReferenceLinksForCollectionNavigationLinkInRequest' and other methods are still in use.